eng/common/scripts/modules/Package-Properties.psm1

```diff
@@ -0,0 +1,177 @@
+# Helper functions for retireving useful information from azure-sdk-for-* repo
+# Example Use : Import-Module .\eng\common\scripts\modules
+class PackageProps
+{
+    [string]$pkgName
+    [string]$pkgVersion
+    [string]$pkgDirectoryPath
+    [string]$pkgReadMePath
+    [string]$pkgChangeLogPath
+
+    PackageProps()
+    {
+    }
+    PackageProps(
+        [string]$pkgName,
+        [string]$pkgVersion,
+        [string]$pkgDirectoryPath
+    )
+    {
+        $this.pkgName = $pkgName
+        $this.pkgVersion = $pkgVersion
+        $this.pkgDirectoryPath = $pkgDirectoryPath
+
+        if (Test-Path (Join-Path $pkgDirectoryPath "README.md"))
+        {
+            $this.pkgReadMePath = Join-Path $pkgDirectoryPath "README.md"
+        } else
+        {
+            $this.pkgReadMePath = $null
+        }
+
+        if (Test-Path (Join-Path $pkgDirectoryPath "CHANGELOG.md"))
+        {
+            $this.pkgChangeLogPath = Join-Path $pkgDirectoryPath "CHANGELOG.md"
+        } else
+        {
+            $this.pkgChangeLogPath = $null
+        }
+    }
+}
+
+function Extract-PkgProps ($pkgPath, $pkgName)
+{
+    if ($Language -eq "net")
+    { Extract-DotNetPkgProps -pkgPath $pkgPath -pkgName $pkgName 
+    }
+    if ($Language -eq "java")
+    { Extract-JavaPkgProps -pkgPath $pkgPath -pkgName $pkgName 
+    }
+    if ($Language -eq "js")
+    { Extract-JsPkgProps -pkgPath $pkgPath -pkgName $pkgName 
+    }
+    if ($Language -eq "python")
+    { Extract-PythonPkgProps -pkgPath $pkgPath -pkgName $pkgName 
+    }
+}
+
+function Extract-DotNetPkgProps ($pkgPath, $pkgName)
+{
+    $projectPath = Join-Path $pkgPath "src" "$pkgName.csproj"
```

Comment:
OK so I think I understand what you were saying about having a matching project name. You are checking it buy file name path. I suppose that will work for now as but we might need to alter that if we ever end up with a case where the project name doesn't match.